package.json

```
@@ -11,7 +11,8 @@
     "lint": "xo",
     "test": "xo && ava",
     "report": "nyc ava",
-    "coverage": "nyc --reporter json --reporter text ava && codecov -f coverage/coverage-final.json"
+    "coverage": "nyc --reporter json --reporter text ava && codecov -f coverage/coverage-final.json",
+    "prepublishOnly": "npx typescript --declaration --emitDeclarationOnly --allowJs index.js"
```

This will use the globally installed version if there is one, or download the latest version. I think it would be better if we could pin a version by adding `typescript` to our `devDependencies`, and then referring to that one here:

```suggestion
    "prepublishOnly": "tsc --declaration --emitDeclarationOnly --allowJs index.js"
```

---

I can pin TypeScript version at npx: `npx typescript@latest ...` if a lower version on a published system is the concern? We need it only before publishing, so, not sure if it better to include and maintain TypeScript as a dependency...